content/en/flux/use-cases/gh-actions-helm-promotion.md

```diff
@@ -155,6 +164,80 @@ The above workflow does the following:
 
 **Note** that you should adapt the workflow to match your release name, namespace and YAML path.
 
+### Promoting a HelmRelease with a digest
+
+The following `.github/workflows/demo-promotion.yaml` workflow will react to the Flux
+repository dispatch events and promote the Helm release to the production cluster.
+
+The expected `HelmRelease` is one with a `OCIRepository` source.
+
+```yaml
+name: demo-promotion
+on:
+  repository_dispatch:
+    types:
+      - HelmRelease/demo.apps
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  promote:
+    runs-on: ubuntu-latest
+    # Start promotion when the staging cluster has successfully
+    # upgraded the Helm release to a new chart version.
+    if: |
+      github.event.client_payload.metadata.env == 'staging' &&
+      github.event.client_payload.severity == 'info'
+    steps:
+      # Checkout main branch.
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+      # Parse the event metadata to determine the chart version deployed on staging.
+      - name: Get chart version from staging
+        id: staging
+        run: |
+          DIGEST=$(echo ${{ github.event.client_payload.metadata.oci-digest }} | cut -d '@' -f1)
+          echo DIGEST=${DIGEST} >> $GITHUB_OUTPUT
+          VERSION=$(echo ${{ github.event.client_payload.metadata.revision }} | cut -d '@' -f1)
+          echo VERSION=${VERSION} >> $GITHUB_OUTPUT
+      # Patch the digest in the production OCIRepository manifest.
+      - name: Set chart version in production
+        id: production
+        env:
+          DIGEST: ${{ steps.staging.outputs.digest }}
+        run: |
+          echo "set ociRepository digest to ${DIGEST}"
+          # This will filter out all resources that do not have a digest field.
+          yq eval '(select(.spec.ref.digest) | .spec.ref.digest) = env(DIGEST)' -i ./clusters/production/apps/demo.yaml
```

Comment:
Let's add a YAML comment after the digest with the chart version. https://mikefarah.gitbook.io/yq/operators/comment-operators#set-line-comment